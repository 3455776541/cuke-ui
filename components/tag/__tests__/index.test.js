import assert from "power-assert";
import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Tag from "../index";

describe("<Tag/>", () => {
	it("should render tag", () => {
		const wrapper = render(
			<div>
				<Tag>黄瓜 ui</Tag>
				<Tag type="primary">黄瓜 ui</Tag>
				<Tag type="info">黄瓜 ui</Tag>
				<Tag type="success">黄瓜 ui</Tag>
				<Tag type="error">黄瓜 ui</Tag>
				<Tag type="warning">黄瓜 ui</Tag>
				<Tag dashed>黄瓜 ui</Tag>
				<Tag disabled>黄瓜 ui</Tag>
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it.skip("should find cuke-tag classnames", () => {
		const wrapper = shallow(
			<div>
				<Tag>黄瓜 ui</Tag>
				<Tag type="primary">黄瓜 ui</Tag>
				<Tag type="info">黄瓜 ui</Tag>
				<Tag type="success">黄瓜 ui</Tag>
				<Tag type="error">黄瓜 ui</Tag>
				<Tag type="warning">黄瓜 ui</Tag>
				<Tag dashed>黄瓜 ui</Tag>
				<Tag disabled>黄瓜 ui</Tag>
			</div>
		);
		assert(wrapper.find(".cuke-tag").length >= 1);
		assert(wrapper.find(".cuke-tag-primary").length === 1);
		assert(wrapper.find(".cuke-tag-info").length === 1);
		assert(wrapper.find(".cuke-tag-success").length === 1);
		assert(wrapper.find(".cuke-tag-error").length === 1);
		assert(wrapper.find(".cuke-tag-warning").length === 1);
		assert(wrapper.find(".cuke-tag-dashed").length === 1);
		assert(wrapper.find(".cuke-tag-disabled").length === 1);
	});
});
