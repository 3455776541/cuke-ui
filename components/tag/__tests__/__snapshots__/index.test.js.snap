// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag/> should render tag 1`] = `
<div>
  <span
    class="cuke-tag cuke-tag-default"
    style="background-color:"
  >
    黄瓜 ui
  </span>
  <span
    class="cuke-tag cuke-tag-primary"
    style="background-color:"
  >
    黄瓜 ui
  </span>
  <span
    class="cuke-tag cuke-tag-info"
    style="background-color:"
  >
    黄瓜 ui
  </span>
  <span
    class="cuke-tag cuke-tag-success"
    style="background-color:"
  >
    黄瓜 ui
  </span>
  <span
    class="cuke-tag cuke-tag-error"
    style="background-color:"
  >
    黄瓜 ui
  </span>
  <span
    class="cuke-tag cuke-tag-warning"
    style="background-color:"
  >
    黄瓜 ui
  </span>
  <span
    class="cuke-tag cuke-tag-default cuke-tag-dashed"
    style="background-color:"
  >
    黄瓜 ui
  </span>
  <span
    class="cuke-tag cuke-tag-default cuke-tag-disabled"
    style="background-color:"
  >
    黄瓜 ui
  </span>
</div>
`;
